keel-sql/src/main/resources/db/changelog/20191108-environment-artifact-constraint.yml

```diff
@@ -0,0 +1,77 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-environment-artifact-constraint-table
+      author: asher
+      changes:
+        - createTable:
+            tableName: environment_artifact_constraint
+            columns:
+              - column:
+                  name: environment_uid
+                  type: char(26)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: artifact_version
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: constraint_type
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: created_at
+                  type: timestamp
+                  defaultValueComputed: CURRENT_TIMESTAMP
+                  constraints:
+                    nullable: false
+              - column:
+                  name: status
+                  type: int
+                  constraints:
+                    nullable: true
+              - column:
+                  name: judged_by
+                  type: varchar(255)
+                  constraints:
+                    nullable: true
+              - column:
+                  name: judged_at
+                  type: timestamp
+                  constraints:
+                    nullable: true
+              - column:
+                  name: comment
+                  type: varchar(255)
+                  constraints:
+                    nullable: true
+              - column:
+                  name: json
+                  type: longtext
+                  constraints:
+                    nullable: true
+
+  - changeSet:
+      id: mysql-environment-artifact-constraint-status-enum
```

Comment:
Why do you create this column as `int` then modify to enum?

Reply:
Liquibase lacks first class enum support, hence the explicit sql statement. Enums are internally stored as ints which is why I used that type as a placeholder.

Reply:
ahh good to know!